databricks/koalas/frame.py

```diff
@@ -239,14 +234,11 @@ def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
             assert not copy
             super(DataFrame, self).__init__(data)
         elif isinstance(data, spark.DataFrame):
+            assert index is None
             assert columns is None
             assert dtype is None
             assert not copy
-            if index is None:
-                super(DataFrame, self).__init__(_InternalFrame(data))
-            else:
-                super(DataFrame, self).__init__(_InternalFrame(
-                    data, data_columns=index.data_columns, index_map=index.index_map))
+            super(DataFrame, self).__init__(_InternalFrame(data))
```

Comment:
Not sure if it's too rare to be a use case, but why not allow passing `index` and `column` to the `_InternalFrame` constructor? `index` should then be a pandas-ish index (i.e. list of scalars) that implicitly needs to be converted to an index map.

Reply:
I'm not sure about the usage of `column` argument, but as for `index`, I think passing pandas-ish index is difficult for Spark backend.

Btw, the `_InternalFrame`'s constructor shouldn't do complex tasks, so we need to prepare it in `DataFrame`'s constructor or prior.